pkg/controller/remotemachineset/awsactuator.go

```diff
@@ -230,6 +237,22 @@ func decodeAWSMachineProviderSpec(rawExt *runtime.RawExtension, scheme *runtime.
 	return spec, nil
 }
 
+func (a *AWSActuator) updateSpotPrice(ms *machineapi.MachineSet, pool *hivev1.MachinePool) error {
+	if pool.Spec.Platform.AWS.SpotMarketOptions == nil {
+		return nil
+	}
+	awsProviderSpec, err := decodeAWSMachineProviderSpec(ms.Spec.Template.Spec.ProviderSpec.Value, a.scheme)
+	if err != nil {
+		return err
+	}
+	awsProviderSpec.SpotMarketOptions = &awsprovider.SpotMarketOptions{
```

Comment:
Can this be done in `updateProviderConfig`?

Reply:
Done